### ISSUE TITLE
Cannot specify a config param in @ConfigMapping via environment variable 

### ISSUE THREAD

**Issue by keitaf** [quarkusio/quarkus#18546]
## Describe the bug

For this config interface,
```
@ConfigMapping(prefix = "test")
public interface TestConfigMapping {
    @WithDefault("1")
    int someValue();
}
```
I cannot specify the value of `someValue` via environment variable.

```
$ TEST_SOME_VALUE=2 ./gradlew quarkusDev
...
2021-07-08 14:20:37,911 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): io.smallrye.config.ConfigValidationException: Configuration validation failed:
        test.some.value does not map to any root
        at io.smallrye.config.ConfigMappingProvider.mapConfiguration(ConfigMappingProvider.java:818)
...

$ TEST_SOMEVALUE=2 ./gradlew quarkusDev
2021-07-08 14:21:29,172 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): io.smallrye.config.ConfigValidationException: Configuration validation failed:
        test.somevalue does not map to any root
        at io.smallrye.config.ConfigMappingProvider.mapConfiguration(ConfigMappingProvider.java:818)
...
```

I can specify it via `-Dtest.some-value={value}`.

(FYI, `@ConfigProperties` works as expected)

### Expected behavior

A config parameter `test.some-value` should be changed with an environment variable `TEST_SOME_VALUE`

### Actual behavior

`ConfigValidationException` is thrown with an error `test.some.value does not map to any root`.

## To Reproduce

1. Create a `@ConfigMapping` interface with a camel-cased parameter name.
2. Try to specify the value of the config parameter via an environment variable by replacing case changes with an underscore and upper-casing all letters.

### Output of `uname -a` or `ver`

Darwin KEFUJII-M-Q636 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "15.0.2" 2021-01-19

### Quarkus version or git rev

2.0.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 6.8.3
## Additional context
(Add any other context about the problem here.)

**Comment by geoand** [comment]
cc @radcortez

**Comment by radcortez** [comment]
Hi @keitaf,

This is already fixed for Quarkus `2.1.0.Final` (not released yet). In the meanwhile, you should be able to work around the issue in the following ways:

Update the Config manually (which fixes the issue):
```xml
    <dependency>
      <groupId>io.smallrye.config</groupId>
      <artifactId>smallrye-config-common</artifactId>
      <version>2.4.1</version>
    </dependency>

    <dependency>
      <groupId>io.smallrye.config</groupId>
      <artifactId>smallrye-config</artifactId>
      <version>2.4.1</version>
    </dependency>
```

Disable the validation:
```properties
smallrye.config.mapping.validate-unknown=false
```

**Comment by keitaf** [comment]
Thank you @radcortez . I'm glad to hear that it'll be fixed in `2.1.0.Final`.
